runtime/jvmti/jvmtiThread.c

```diff
@@ -1072,12 +1078,29 @@ resumeThread(J9VMThread * currentThread, jthread thread)
 	J9VMThread *targetThread = NULL;
 	jvmtiError rc = getVMThread(currentThread, thread, &targetThread, FALSE, TRUE);
 	if (rc == JVMTI_ERROR_NONE) {
-		if (targetThread->publicFlags & J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND) {
-			clearHaltFlag(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
-			Trc_JVMTI_threadResumed(targetThread);
-		} else {
-			rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
+#if JAVA_SPEC_VERSION >= 19
+		j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+		if (NULL != targetThread)
+#endif /* JAVA_SPEC_VERSION >= 19 */
+		{
+			if (OMR_ARE_ANY_BITS_SET(targetThread->publicFlags, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND)) {
+				clearHaltFlag(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
+				Trc_JVMTI_threadResumed(targetThread);
+			} else {
+				rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
+			}
+		}
+#if JAVA_SPEC_VERSION >= 19
+		else if (IS_VIRTUAL_THREAD(currentThread, threadObject)) {
+			/* targetThread is NULL only for unmounted virtual threads. */
+			jint vthreadState = J9VMJAVALANGVIRTUALTHREAD_STATE(currentThread, threadObject);
+			if (OMR_ARE_NO_BITS_SET(vthreadState, JVMTI_VTHREAD_STATE_SUSPENDED)) {
+				rc = JVMTI_ERROR_THREAD_NOT_SUSPENDED;
+			} else {
+				J9VMJAVALANGVIRTUALTHREAD_SET_STATE(currentThread, threadObject, vthreadState & ~JVMTI_VTHREAD_STATE_SUSPENDED);
+			}
```

Comment:
I suggest we should have an `else` block so we don't return `JVMTI_ERROR_NONE` if the thread is not virtual.

Reply:
getVMThread has an assert which guarantees that targetThread is NULL only for virtual threads when mustBeAlive is TRUE. The `else if` can just be changed to an `else`.
https://github.com/eclipse-openj9/openj9/blob/f06248f89427d0509360ec6a1df74e0c06436072/runtime/jvmti/jvmtiHelpers.c#L178